crypto/asn1/t_pkey.c

```diff
@@ -69,22 +69,16 @@ int ASN1_bn_print(BIO *bp, const char *number, const BIGNUM *num,
         return 1;
     }
 
-    buflen = BN_num_bytes(num) + 1;
-    buf = tmp = OPENSSL_malloc(buflen);
+    buflen = BN_num_bytes(num);
+    buf = OPENSSL_malloc(buflen);
```

Comment:
And this will allocate 0 bytes when BN is zero, so malloc is free to return NULL or not.

Reply:
oops forget it, I overlooked the BN_is_zero above...